server/src/main/java/org/elasticsearch/rest/MethodHandlers.java

```diff
@@ -56,11 +59,29 @@ MethodHandlers addMethods(RestHandler handler, RestRequest.Method... methods) {
     }
 
     /**
-     * Returns the handler for the given method or {@code null} if none exists.
+     * Return a handler for a given method and a version
+     * If a handler for a given version is not found, the handler for Version.CURRENT is returned.
+     * We only expect Version.CURRENT or Version.CURRENT -1. This is validated.
+     *
+     * Handlers can be registered under the same path and method, but will require to have different versions (CURRENT or CURRENT-1)
+     *
+     * //todo What if a handler was added in V8 but was not present in V7?
```

Comment:
IMO if you request V7 compatibility mode but the API doesn't exist in V7 then this should be a bad request (HTTP 400)